src/Bandits/BanditReferenceIndexer.php

```diff
@@ -2,20 +2,23 @@
 
 namespace Eppo\Bandits;
 
-use Eppo\DTO\Bandit\BanditVariation;
+use Eppo\DTO\Bandit\BanditFlagVariation;
+use Eppo\DTO\BanditReference;
 use Eppo\Exception\InvalidConfigurationException;
 
-class BanditVariationIndexer implements IBanditVariationIndexer
+class BanditReferenceIndexer implements IBanditReferenceIndexer
 {
-    // By just serializing the indexed variations, we cut down on cache size.
+    // magic methods to store just the underlying data when serialized to cache.
     public function __serialize(): array
     {
-        return $this->banditFlags;
+        return [
+            'flagIndex' => $this->flagIndex
```

Comment:
future work will have multiple private fields stored when serialized. Using a map here prepares for this.